src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -609,10 +609,17 @@ private static AuthInterceptorContext CreateAuthInterceptorContext(Uri baseAddre
             _writeCompleteTcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
             message.Content = new PushStreamContent(
-                (stream) =>
+                async stream =>
                 {
+                    // Immediately flush request stream to send headers
+                    // https://github.com/dotnet/corefx/issues/39586#issuecomment-516210081
+                    await stream.FlushAsync().ConfigureAwait(false);
```

Comment:
Is this why it hangs?!

Reply:
If there is no HTTP traffic being sent from the HttpClient at that time, yes.